client/my-sites/controller.js

```diff
@@ -39,21 +39,26 @@ const sites = sitesFactory();
 /*
  * The main navigation of My Sites consists of a component with
  * the site selector list and the sidebar section items
+ * @param { object } context - Middleware context
+ * @returns { object } React element containing the site selector and sidebar
  */
-function renderNavigation( context, allSitesPath, siteBasePath ) {
-	// Render the My Sites navigation in #secondary
-	ReactDom.render(
-		React.createElement( ReduxProvider, { store: context.store },
-			React.createElement( NavigationComponent, {
-				layoutFocus,
-				path: context.path,
-				allSitesPath,
-				siteBasePath,
-				user,
-				sites
-			} )
-		),
-		document.getElementById( 'secondary' )
+function createNavigation( context ) {
+	const siteFragment = route.getSiteFragment( context.pathname );
+	let basePath = context.pathname;
+
+	if ( siteFragment ) {
+		basePath = route.sectionify( context.pathname );
+	}
+
+	return React.createElement( ReduxProvider, { store: context.store },
```

Comment:
We should really just start using JSX in these files. Not sure why we stuck with `React.createElement`